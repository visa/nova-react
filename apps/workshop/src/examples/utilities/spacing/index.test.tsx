/**
 *              Â© 2025 Visa
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import metaData from './meta.json';

import { GapColumnSpacing } from './gap-column-spacing';
import { GapInheritSpacing } from './gap-inherit-spacing';
import { GapNormalSpacing } from './gap-normal-spacing';
import { GapRowSpacing } from './gap-row-spacing';
import { GapSpacing } from './gap-spacing';
import { MarginBottomSpacing } from './margin-bottom-spacing';
import { MarginInheritSpacing } from './margin-inherit-spacing';
import { MarginHorizontalSpacing } from './margin-horizontal-spacing';
import { MarginLeftSpacing } from './margin-left-spacing';
import { MarginRightSpacing } from './margin-right-spacing';
import { DefaultMargin } from './default-margin';
import { MarginTopSpacing } from './margin-top-spacing';
import { MarginVerticalSpacing } from './margin-vertical-spacing';
import { PaddingBottomSpacing } from './padding-bottom-spacing';
import { PaddingHorizontalSpacing } from './padding-horizontal-spacing';
import { PaddingInheritSpacing } from './padding-inherit-spacing';
import { PaddingLeftSpacing } from './padding-left-spacing';
import { PaddingRightSpacing } from './padding-right-spacing';
import { DefaultPadding } from './default-padding';
import { PaddingTopSpacing } from './padding-top-spacing';
import { PaddingVerticalSpacing } from './padding-vertical-spacing';

const examples = [
  { Component: GapColumnSpacing, title: metaData['gap-column-spacing'].title },
  { Component: GapInheritSpacing, title: metaData['gap-inherit-spacing'].title },
  { Component: GapNormalSpacing, title: metaData['gap-normal-spacing'].title },
  { Component: GapRowSpacing, title: metaData['gap-row-spacing'].title },
  { Component: GapSpacing, title: metaData['gap-spacing'].title },
  { Component: MarginBottomSpacing, title: metaData['margin-bottom-spacing'].title },
  { Component: MarginHorizontalSpacing, title: metaData['margin-horizontal-spacing'].title },
  { Component: MarginLeftSpacing, title: metaData['margin-left-spacing'].title },
  { Component: MarginRightSpacing, title: metaData['margin-right-spacing'].title },
  { Component: DefaultMargin, title: metaData['default-margin'].title },
  { Component: PaddingHorizontalSpacing, title: metaData['padding-horizontal-spacing'].title },
  { Component: PaddingInheritSpacing, title: metaData['padding-inherit-spacing'].title },
  { Component: PaddingLeftSpacing, title: metaData['padding-left-spacing'].title },
  { Component: PaddingRightSpacing, title: metaData['padding-right-spacing'].title },
  { Component: PaddingTopSpacing, title: metaData['padding-top-spacing'].title },
  { Component: PaddingVerticalSpacing, title: metaData['padding-vertical-spacing'].title },
  { Component: MarginInheritSpacing, title: metaData['margin-inherit-spacing'].title },
  { Component: MarginTopSpacing, title: metaData['margin-top-spacing'].title },
  { Component: MarginVerticalSpacing, title: metaData['margin-vertical-spacing'].title },
  { Component: PaddingBottomSpacing, title: metaData['padding-bottom-spacing'].title },
  { Component: DefaultPadding, title: metaData['default-padding'].title },
];

describe('Spacing examples', () => {
  examples.forEach(({ Component, title }) => {
    it(`${title} should render correctly`, async () => {
      const { container } = render(<Component />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
      expect(container).toMatchSnapshot();
    });
  });
});
