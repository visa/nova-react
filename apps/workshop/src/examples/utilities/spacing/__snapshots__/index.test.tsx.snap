// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacing examples Column gap should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-col-gap-20 v-flex"
    >
      <div
        class="v-flex v-flex-col"
      >
        <div
          class="v-surface v-flex-grow v-p-16"
          style="box-shadow: var(--elevation-medium); inline-size: auto;"
        >
          Item 1
        </div>
        <div
          class="v-surface v-flex-grow v-p-16"
          style="box-shadow: var(--elevation-medium); inline-size: auto;"
        >
          Item 3
        </div>
      </div>
      <div
        class="v-flex v-flex-col"
      >
        <div
          class="v-surface v-flex-grow v-p-16"
          style="box-shadow: var(--elevation-medium); inline-size: auto;"
        >
          Item 2
        </div>
        <div
          class="v-surface v-flex-grow v-p-16"
          style="box-shadow: var(--elevation-medium); inline-size: auto;"
        >
          Item 4
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Default gap should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-col-gap-20 v-flex"
    >
      <div
        class="v-surface v-flex-grow v-p-16"
        style="box-shadow: var(--elevation-medium); inline-size: auto;"
      >
        Item 1
      </div>
      <div
        class="v-surface v-flex-grow v-p-16"
        style="box-shadow: var(--elevation-medium); inline-size: auto;"
      >
        Item 2
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Default margin should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-flex"
    >
      <div
        class="v-surface v-m-20"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Gap inherit should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <ul
      class="v-flex v-flex-col v-gap-8"
    >
      <li
        class="v-align-items-center v-flex v-gap-inherit"
      >
        <span
          aria-label="Alex Miller"
          class="v-avatar v-avatar-small"
        >
          AM
        </span>
        Alex Miller
      </li>
      <li
        class="v-align-items-center v-flex v-gap-inherit"
      >
        <span
          aria-label="Rosetta Jones"
          class="v-avatar v-avatar-small"
        >
          RJ
        </span>
        Rosetta Jones
      </li>
      <li
        class="v-align-items-center v-flex v-gap-inherit"
      >
        <span
          aria-label="Stacey Taylor"
          class="v-avatar v-avatar-small"
        >
          ST
        </span>
        Stacey Taylor
      </li>
    </ul>
  </div>
</div>
`;

exports[`Spacing examples Gap normal should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-flex v-gap-normal"
    >
      <div
        class="v-surface v-flex-grow v-p-16"
        style="box-shadow: var(--elevation-medium); inline-size: auto;"
      >
        Item 1
      </div>
      <div
        class="v-surface v-flex-grow v-p-16"
        style="box-shadow: var(--elevation-medium); inline-size: auto;"
      >
        Item 2
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Margin bottom should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-flex"
    >
      <div
        class="v-surface v-mb-20 v-p-16"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Margin horizontal should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-flex"
    >
      <div
        class="v-surface v-mx-20 v-p-16"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Margin inherit should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div>
      <div
        class="v-mt-20"
        style="text-align: center;"
      >
        <p>
          Parent with top margin
        </p>
        <div
          class="v-surface v-mt-inherit v-p-16"
          style="border-inline: 0; border-block-end: 0;"
        >
          Child inherits top margin from parent
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Margin left should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-flex"
    >
      <div
        class="v-surface v-ml-20 v-p-16"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Margin right should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-flex"
    >
      <div
        class="v-surface v-mr-20 v-p-16"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Margin top should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-flex"
    >
      <div
        class="v-surface v-mt-20 v-p-16"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Margin vertical should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-flex"
    >
      <div
        class="v-surface v-my-20 v-p-16"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Padding bottom should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-pb-20"
    >
      <div
        class="v-surface v-p-16 v-pb-20"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Padding horizontal should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-px-20"
    >
      <div
        class="v-surface v-p-16 v-pb-20"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Padding inherit should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-pt-24"
      style="text-align: center;"
    >
      <p>
        Parent with top padding
      </p>
      <div
        class="v-surface v-p-16 v-pt-inherit"
        style="border-inline: 0; border-block-end: 0;"
      >
        Child inherits top margin from parent
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Padding left should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-pl-20"
    >
      <div
        class="v-surface v-p-16 v-pb-20"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Padding right should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-pr-20"
    >
      <div
        class="v-surface v-p-16 v-pb-20"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Padding should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-p-20"
    >
      <div
        class="v-surface v-p-16 v-pb-20"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Padding top should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-pt-20"
    >
      <div
        class="v-surface v-p-16 v-pb-20"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Padding vertical should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-py-20"
    >
      <div
        class="v-surface v-p-16 v-pb-20"
      >
        Content Area
      </div>
    </div>
  </div>
</div>
`;

exports[`Spacing examples Row gap should render correctly 1`] = `
<div>
  <div
    class="v-flex"
  >
    <div
      class="v-flex v-flex-col v-row-gap-20"
    >
      <div
        class="v-flex"
      >
        <div
          class="v-surface v-flex-grow v-p-16"
          style="box-shadow: var(--elevation-medium); inline-size: auto;"
        >
          Item 1
        </div>
        <div
          class="v-surface v-flex-grow v-p-16"
          style="box-shadow: var(--elevation-medium); inline-size: auto;"
        >
          Item 2
        </div>
      </div>
      <div
        class="v-flex"
      >
        <div
          class="v-surface v-flex-grow v-p-16"
          style="box-shadow: var(--elevation-medium); inline-size: auto;"
        >
          Item 3
        </div>
        <div
          class="v-surface v-flex-grow v-p-16"
          style="box-shadow: var(--elevation-medium); inline-size: auto;"
        >
          Item 4
        </div>
      </div>
    </div>
  </div>
</div>
`;
